### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by owinter86**
- Move codepush variables from package.json script to the codepush node script, I think this is easier to manage while the script is maintained in project file, if externalised as a package then it would make sense to have it pass in variables via the scripts.
- Set ruby versions and gemset wrapper (ensures all machines are using the same gemsets internally)
- Add fastlane with dsym symbol upload lane which is required for dsym symbol upload for bitcode enabled apps
- Add basic readme with your M1 environment requirements

### CODE REVIEW

File: template/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+- `yarn ios` builds against staging environment to the ios simulator with the React Native debugging tools attached allowing flipper to be used.
+- `yarn iosStaging` builds release app (no dev tools attached) that is targetting the STAGING environment
+- `yarn iosRelease` builds release app (no dev tools attached) that is targetting the PRODUCTION environment
+- `yarn ios --device "NAME OF DEVICE"` This will build to the ios device plugged into the machine - this requires AD HOC profivisioning profiles setup in the signing and configuration section of xcode and installed on the developers machine.
+
+## Building Android target
+
+```bash
+bundle install
```
Comment (dov11) [inline template/README.md:47]: Is that needed for android?

File: template/appcenter-post-build.sh [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Generate sourcemaps for Bugsnag
 
-# REPLACE <MYAPP_NAME> with your app name and <MY_GITHUB_REPO_SSH> with your repo ssh
+# REPLACE <MY_GITHUB_REPO_SSH> with your repo ssh
 
 BUILD_NUMBER=$(git show -s --format=%ct)
-VERSION=$(/usr/libexec/PlistBuddy -c "Print :CFBundleShortVersionString" ios/<MYAPP_NAME>/info.plist)
+VERSION=$(/usr/libexec/PlistBuddy -c "Print :CFBundleShortVersionString" ios/MyApp/info.plist)
```
Comment (dov11) [inline template/appcenter-post-build.sh:6]: Is `MyApp` in appcenter script getting picked up and replaced by the template?

**Review by dov11 — APPROVED** [review]

File: template/appcenter-post-build.sh (hunk 2 above)
Comment (owinter86) [reply template/appcenter-post-build.sh:6]: yep

Comment (dov11) [reply template/appcenter-post-build.sh:6]: nice

File: template/README.md (hunk 1 above)
Comment (owinter86) [reply template/README.md]: Nope, I have removed it

**Pull request merged**